sinks/influxdb/driver.go

```diff
@@ -0,0 +1,200 @@
+// Copyright 2015 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package influxdb
+
+import (
+	"fmt"
+	"sort"
+	"strings"
+	"sync"
+	"time"
+
+	sink_api "github.com/GoogleCloudPlatform/heapster/sinks/api"
+	"github.com/golang/glog"
+	influxdb "github.com/influxdb/influxdb/client"
+)
+
+type influxdbSink struct {
+	hostname  string
+	database  string
+	client    *influxdb.Client
+	series    []*influxdb.Series
+	dbName    string
+	stateLock sync.RWMutex
+	// TODO(rjnagal): switch to atomic if writeFailures is the only protected data.
+	writeFailures int // guarded by stateLock
+	avoidColumns  bool
+}
+
+func (self *influxdbSink) Register(metrics []sink_api.MetricDescriptor) error {
+	// Create tags once influxDB v0.9.0 is released.
+	return nil
+}
+
+func (self *influxdbSink) labelsToString(labels map[string]string) string {
```

Comment:
Don't we have another version of this in API? Lets make it public and use it here. Or place it in util.